### ISSUE TITLE
fix(ui): drop hardcoded backend option sets from JS (DataTab CV_STRATEGIES + ModelEditors LGBM keys)

### ISSUE THREAD

**Issue by nbx-liz** [nbx-liz/LizyML-Widget#119]
## Summary

CLAUDE.md 4 and 8 explicitly forbid the following (paraphrased):

- 8: JS must not hardcode backend-specific option sets, parameter catalogs, or step values
- 4: the UI layer reads/writes traitlets and renders `backend_contract` generically; it must not call Python or ML libraries directly, and must not hold backend-specific option sets, parameter catalogs, or step values

Two violations slipped through review and are still on `develop`. They make the widget brittle against future backend changes — Python-side updates to `backend_contract.capabilities.*` have no effect on the parts of the UI that hardcode the same data.

### M-2: `CV_STRATEGIES` hardcoded in DataTab

[js/src/tabs/DataTab.tsx:44-53](js/src/tabs/DataTab.tsx#L44-L53):

```typescript
const CV_STRATEGIES = [
  { value: "kfold", label: "KFold" },
  { value: "stratified_kfold", label: "StratifiedKFold" },
  { value: "group_kfold", label: "GroupKFold" },
  { value: "stratified_group_kfold", label: "StratifiedGroupKFold" },
  { value: "time_series", label: "TimeSeriesSplit" },
  { value: "group_time_series", label: "GroupTimeSeriesSplit" },
  { value: "blocked_group_kfold", label: "BlockedGroupKFold" },
];
```

Meanwhile `backend_contract.capabilities.cv_strategies` already ships the same list from Python. The behavioural-side derivations in this file (`FALLBACK_NEEDS_GROUP` etc.) correctly read from the contract; only the display side is stale.

**Fix**: derive `CV_STRATEGIES` from `backendContract?.capabilities?.cv_strategies`. Labels can come from a small label-map (also delivered via contract `ui_schema` if needed) or a fallback `humanise(value)` helper.

### M-3: LightGBM-specific catalog hardcoded in ModelEditors

[js/src/components/ModelEditors.tsx:14-17, 139, 142, 339, 374](js/src/components/ModelEditors.tsx):

- `HANDLED_MODEL_FIELDS` enumerates LightGBM-specific keys (`auto_num_leaves`, `num_leaves_ratio`, `feature_weights`, `balanced`)
- `num_leaves` default literal `256` (a LightGBM constant) hardcoded in the UI default
- Manual list of hint keys including `verbose`, `num_threads`

These are exactly the "no step values / no defaults / no LGBM-specific keys in JS" rule from CLAUDE.md 4 leaking through. If the Adapter's `parameter_hints` or `option_sets` is updated tomorrow to deprecate `num_leaves` in favour of a backend-driven equivalent, the JS will not reflect it.

**Fix**:
- Move the LightGBM-specific keys into `backend_contract.ui_schema.parameter_hints` (the existing channel) so JS reads them generically
- Drop the `256` literal — the default belongs in the backend's config initialisation (it already does at `adapter.py:initialize_config`); JS should render whatever value the contract+config currently holds
- Replace the hint-key allowlist with an iteration over `parameter_hints` from the contract

## Acceptance criteria

- [ ] `js/src/tabs/DataTab.tsx` no longer contains `CV_STRATEGIES` literal; the dropdown is populated from `backend_contract.capabilities.cv_strategies` (with a humanise-fallback for labels if `ui_schema` does not provide them)
- [ ] `js/src/components/ModelEditors.tsx` no longer references LightGBM-specific keys, `num_leaves=256`, or any model-specific hint key by name; everything routes through `backend_contract.ui_schema.parameter_hints`
- [ ] Grep clean: `git grep -nE "kfold|stratified|num_leaves|num_threads" js/src/` returns only references that are dynamic (template-derived from contract data) or test fixtures
- [ ] Existing JS unit tests pass; new tests verify:
  - DataTab renders an unknown CV strategy if the backend reports one (catches the static-list regression)
  - ModelEditors renders an unknown model param key if the backend reports it
- [ ] Adding a new CV strategy in `adapter_contract.py` automatically appears in the UI dropdown without any JS change (verified by extending an existing test)

## Why this is worth doing

The widget's value proposition (per BLUEPRINT.md) is "backend-driven UI". These two files break that promise — they are the only two places where adding a new CV strategy or model param requires a JS-side edit. Closing them out makes future backend additions (a sklearn or XGBoost adapter, for example) a Python-only change.

## Out of scope

- Migrating the `parameter_hints` schema itself — only the consumer side changes here
- New backend support — separate proposal

Refs: code review thread on develop after P-030 (#113); CLAUDE.md 4, 8.

**Comment by nbx-liz** [comment]
Resolved via #121.
